serialization/src/main/kotlin/io/kinference/protobuf/arrays/TiledArrayContainer.kt

```diff
@@ -0,0 +1,41 @@
+package io.kinference.protobuf.arrays
+
+import io.kinference.ndarray.extensions.createArray
+import io.kinference.protobuf.ProtobufReader
+import io.kinference.protobuf.arrays.PrimitiveArraySerializer.Companion.arraySerializer
+import io.kinference.protobuf.arrays.TiledArraySerializer.Companion.tiledSerializer
+import io.kinference.protobuf.message.TensorProto
+
+internal class TiledArrayContainer {
+    private var array: Any? = null
+    private var tiled: Any? = null
+
+    private var tiledInitialized: Boolean = false
+
+    fun hasData() = array != null || tiled != null
+
+    fun setTiled(array: Any) {
+        tiled = array
+        tiledInitialized = true
+    }
+
+    fun decode(reader: ProtobufReader, tag: Int, dataType: TensorProto.DataType, shape: IntArray?) {
```

Comment:
In which cases we don't have shape?

Reply:
When we read the data, but shape wasn't initialized yet